python/pyspark/sql/tests/test_types.py

```diff
@@ -118,8 +118,9 @@ def test_infer_schema(self):
 
         with self.tempView("test"):
             df.createOrReplaceTempView("test")
-            result = self.spark.sql("SELECT l[0].a from test where d['key'].d = '2'")
-            self.assertEqual(1, result.head()[0])
+            with self.sql_conf({"spark.sql.ansi.enabled": False}):
```

Comment:
Can you explain why we set this as `False`, and can we create another DataFrame that fails only when `spark.sql.ansi.enabled` is `False` instead of disabling it for all?